### PR TITLE
Cscli Explain Stdin

### PR DESCRIPTION
**Pull request by LaurenceJJones**
#1594 
Allow user to provide stdin to ```cscli explain```:
```bash
tail -n 10 /var/log/nginx/access.log | cscli explain --type nginx
head -n 10 /var/log/nginx/access.log | cscli explain --type nginx
cscli explain --type nginx < /var/log/nginx/access.log
```

### CODE REVIEW

**Comment by mmetc** [comment]
Thanks

Maybe we can implement reading from stdin without temporary files, by wrapping it in a datasource that is then picked up by pkg/acquisition/modules/file. I don't know if we are following a specific standard for the dsn string

**Comment by AlteredCoder** [comment]
Hello,

To be more compliant with other programs, can we use `-f -` to specify that we read from stdin please?
For example:

```
cat log.file.txt | cscli explain --type nginx -f -

**Review by AlteredCoder — CHANGES_REQUESTED** [review]
Hello,

To be more compliant with other programs, can we use `-f -` to specify that we read from stdin please?
For example:

```
cat logfile.txt | cscli explain --type nginx -f -

File: cmd/crowdsec-cli/explain.go [hunk 1]
```diff
@@ -85,11 +115,11 @@ cscli explain --dsn "file://myfile.log" --type nginx
 				log.Fatalf("fail to run crowdsec for test: %v", err)
 			}
 
-			// rm the temporary log file if only a log line was provided
-			if logLine != "" {
+			// rm the temporary log file if only a log line/stdin was provided
+			if tmpFile != "" {
 				f.Close()
```
Comment (LaurenceJJones) [inline cmd/crowdsec-cli/explain.go:120]: Hey! thanks @buixor ! just wondering if we need to close the file handler twice?

Comment (AlteredCoder) [reply cmd/crowdsec-cli/explain.go:120]: yes i think the file is already close before

**Pull request merged**